gating/update_dependencies/run

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash -xe
+
+# Update OSA SHA to head of stable/ocata
+
+# Get current head of stable/ocata
+git clone "https://github.com/openstack/openstack-ansible" ${WORKSPACE}/oa
+pushd ${WORKSPACE}/oa
+  git checkout stable/ocata
+  CURRENT_SHA="$(git log -n 1 --format=%H)"
+popd
```

Comment:
This is also going to need to update the value of rpc_release if the current value matches the latest tag on this branch.

Reply:
Something like this:

```
RPC_RELEASE="$(/opt/rpc-openstack/scripts/artifacts-building/derive-artifact-version.sh)"
LATEST_RPC_RELEASE="$(git describe --abbrev=0 --tags)"

if [[ ${RPC_RELEASE} == ${LATEST_RPC_RELEASE} ]]; then
      echo "Updating the release version..."

      # Extract the required version info
      major_version=$( echo ${currentversion} | cut -d. -f1 )
      minor_version=$( echo ${currentversion} | cut -d. -f2 )
      patch_version=$( echo ${currentversion} | cut -d. -f3 )

      # increment the minor version
      minor_version=$(( minor_version + 1 ))

      sed -i "s/${RPC_RELEASE}/${major_version}.${minor_version}.${patch_version}/" group_vars/all/release.yml
fi
```

Reply:
The code seem ok to me.

Reply:
This doesn't sound like a dependency update, this sounds like it should be part of the release hook.

Reply:
I just realised that the latest tag would be two versions behind the development branch:

```
newton -> rpc_release = 14.2.0
newton-rc -> rpc_release = 14.1.0
latest tag -> rpc_release = 14.0.0
```

This means that the previously suggested script won't work. We'd have to read the rpc_release version from the rc branch. :/

@git-harry You're probably right, except that this is an in-repo change which requires a PR. It's a quirk of this particular repository so one could argue that it better fits being done in this hook. However I would not object to some sort of post-release job proposing some sort of change to the repo in some kind of post release hook.